### ISSUE TITLE
ImageInspectResponse.GraphDriver.Data is not a String

### ISSUE THREAD

**Issue by davidtnorth** [docker-java/docker-java#538]
The following code:

> 
public static void main(final String[] args) throws Exception {
    Properties properties = new Properties();
    properties.put("DOCKER_TLS_VERIFY", "0");
    properties.put("DOCKER_HOST", "tcp://localhost:2375");
    DockerClientConfig clientCFG = new DockerClientConfig.DockerClientConfigBuilder()
        .withProperties(properties)
        .build();
    DockerClient dockerClient = DockerClientBuilder.getInstance(clientCFG).build();
    dockerClient.pullImageCmd("busybox:latest").exec(new PullImageResultCallback()).awaitCompletion();
    final InspectImageResponse inspectImageResponse = dockerClient.inspectImageCmd("busybox:latest").exec();
  }

works in 3.0-RC3 and fails in 3.0-RC4 as follows:
```
Exception in thread "main" javax.ws.rs.ProcessingException: Error reading entity from input stream.
	at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:866)
	at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:783)
	at org.glassfish.jersey.client.ClientResponse.readEntity(ClientResponse.java:326)
	at org.glassfish.jersey.client.JerseyInvocation.translate(JerseyInvocation.java:786)
	at org.glassfish.jersey.client.JerseyInvocation.access$500(JerseyInvocation.java:91)
	at org.glassfish.jersey.client.JerseyInvocation$2.call(JerseyInvocation.java:683)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:228)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:424)
	at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:679)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:408)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.get(JerseyInvocation.java:308)
	at com.github.dockerjava.jaxrs.InspectImageCmdExec.execute(InspectImageCmdExec.java:27)
	at com.github.dockerjava.jaxrs.InspectImageCmdExec.execute(InspectImageCmdExec.java:13)
	at com.github.dockerjava.jaxrs.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:23)
	at com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at com.github.dockerjava.core.command.InspectImageCmdImpl.exec(InspectImageCmdImpl.java:40)
	at InspectBug.main(InspectBug.java:22)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.lang.String out of START_OBJECT token
 at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream@1b7cc17c; line: 1, column: 1157] (through reference chain: com.github.dockerjava.api.command.InspectImageResponse["GraphDriver"]->com.github.dockerjava.api.command.GraphDriver["Data"])
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:148)
	at com.fasterxml.jackson.databind.DeserializationContext.mappingException(DeserializationContext.java:857)
	at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:62)
	at com.fasterxml.jackson.databind.deser.std.StringDeserializer.deserialize(StringDeserializer.java:11)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:520)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:101)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:258)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:520)
	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:101)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:258)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
	at com.fasterxml.jackson.databind.ObjectReader._bind(ObjectReader.java:1470)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:912)
	at com.fasterxml.jackson.jaxrs.base.ProviderBase.readFrom(ProviderBase.java:811)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.invokeReadFrom(ReaderInterceptorExecutor.java:264)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:234)
	at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:154)
	at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1124)
	at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:851)
	... 18 more
```

**Comment by KostyaSha** [comment]
Please enable unsecured HTTP API connection to daemon and run wireshark on your client. Then run sniffing, run this code, stop sniffing, click "follow tcp stream" and then post to use request and response data. That would allow identify and wrote test for your response.

**Comment by marcuslinke** [comment]
@davidtnorth What docker version do you use?

**Comment by davidtnorth** [comment]
Docker version 1.10.2, build c3959b1

**Comment by marcuslinke** [comment]
@davidtnorth Could you enable debug logging by putting this [file](https://github.com/docker-java/docker-java/blob/master/src/test/resources/logback.xml) into your classpath and dump the json response please?

**Comment by marcuslinke** [comment]
Format is described here: https://github.com/docker/docker/blob/master/man/docker-inspect.1.md

**Comment by KostyaSha** [comment]
Test available.
New json sample need to be added and 
```
"GraphDriver": {
    "Name": "devicemapper",
    "Data": {
        "DeviceId": "5",
        "DeviceName": "docker-253:1-2763198-d2cc496561d6d520cbc0236b4ba88c362c446a7619992123f11c809cded25b47",
        "DeviceSize": "171798691840"
    }
},
``` 
should be tested

**Comment by KostyaSha** [comment]
@marcuslinke should we use some `JsonObject` instead `Object` for unknown fields?

**Comment by marcuslinke** [comment]
I don't think so. JsonObject is for complex objects only, so what if the unknown type is String?

**Comment by KostyaSha** [comment]
Will it be String but in Json raw object?

**Comment by KostyaSha** [comment]
Or mapper guesses itself?

**Comment by marcuslinke** [comment]
If we use `Object` the mapper accepts any json type for the field. I guess when we use someting like `JsonObject` it refers to a complex json object aka a Map in java, so it will fail when json contains a string value for the field.

**Comment by KostyaSha** [comment]
I didn't found any docs so decided that unknown field could be text, that's why used String. But it my mistake :(

**Comment by marcuslinke** [comment]
Beside this I don't find any jackson `JsonObject` class. I was confused by another json library I use at work. So what `JsonObject` do you mean?

**Comment by marcuslinke** [comment]
However could we find the unknown type fields? I guess these are `null` in sample json, right?

**Comment by KostyaSha** [comment]
fix #541

**Comment by KostyaSha** [comment]
> However could we find the unknown type fields? I guess these are null in sample json, right?

I mistakenly thought that it failed for Object, but here was String. Bug is obvious.
